test/extended/images/helper.go

```diff
@@ -423,3 +588,127 @@ func IsBlobStoredInRegistry(
 	}
 	return present, presentInRepository, err
 }
+
+// RunHardPrune executes inside a docker-registry pod and runs a garbage collector. The docker-registry is
+// assumed to be in a read-only mode and using filesystem as a storage driver. It returns lists of deleted
+// files.
+func RunHardPrune(oc *exutil.CLI) (*RegistryStorageFiles, error) {
+	podList, err := oc.AsAdmin().KubeClient().Core().Pods("default").List(metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{"deploymentconfig": "docker-registry"}).String(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(podList.Items) == 0 {
+		return nil, fmt.Errorf("failed to find any docker-registry pod")
+	}
+
+	defer func(ns string) { oc.SetNamespace(ns) }(oc.Namespace())
+	output, err := oc.AsAdmin().SetNamespace(metav1.NamespaceDefault).Run("env").Args("--list", "dc/docker-registry").Output()
+	if err != nil {
+		return nil, err
+	}
+
+	podName := podList.Items[0].Name
+	deleted := &RegistryStorageFiles{
+		Repos:         []string{},
+		ManifestLinks: make(RepoLinks),
+		LayerLinks:    make(RepoLinks),
+		Blobs:         []string{},
+	}
+
+	err = wait.ExponentialBackoff(retry.DefaultBackoff, func() (bool, error) {
+		out, err := oc.SetNamespace(metav1.NamespaceDefault).AsAdmin().
+			Run("exec").Args("--stdin", podName, "--", "/bin/sh", "-s").
+			InputString(fmt.Sprintf(registryGCLauncherScript, dockerRegistryBinary)).Output()
+		if exitError, ok := err.(*exutil.ExitError); ok && strings.Contains(exitError.StdErr, "unable to upgrade connection") {
```

Comment:
Looks like there isn't a container `registry` in the `docker-registry-1-z3hj1` pod? I haven't seen this before but it may mean the registry failed to start.